snuba/admin/static/tracing/index.tsx

```diff
@@ -1,8 +1,115 @@
-import React from "react";
+import React, { useState } from "react";
 import Client from "../api_client";
 import QueryDisplay from "../components/query_display/query_display";
 import { QueryResult } from "../components/query_display/types";
 
+type LogLine = {
+  host: string;
+  pid: string;
+  query_id: string;
+  log_level: string;
+  component: string;
+  message: string;
+};
+
+type BucketedLogs = Map<String, Map<MessageCategory, LogLine[]>>;
+
+const logLineMatcher =
+  /\[ (?<hostname>\S+) \] \[ (?<pid>\d+) \] \{(?<local_query_id>[^}]+)\} <(?<log_level>[^>]+)> (?<component>[^:]+): (?<message>.*)/;
+
+function parseLogLine(logLine: string): LogLine | null {
+  const logLineRegexMatch = logLine.match(logLineMatcher);
+  const host = logLineRegexMatch?.groups?.hostname;
+  const pid = logLineRegexMatch?.groups?.pid;
+  const local_query_id = logLineRegexMatch?.groups?.local_query_id;
+  const log_level = logLineRegexMatch?.groups?.log_level;
+  const message = logLineRegexMatch?.groups?.message;
+  const component = logLineRegexMatch?.groups?.component;
+
+  if (host && pid && local_query_id && log_level && message) {
+    const logLineParsed: LogLine = {
+      host: host,
+      pid: pid!,
+      query_id: local_query_id,
+      log_level: log_level!,
+      component: component!,
+      message: message!,
+    };
+    return logLineParsed;
+  } else {
+    return null;
```

Comment:
Which are the lines that can't be formatted this way - is it common? I wonder if we should display the output instead of removing them in case they contain something useful.

Reply:
I think the trace output contains one empty line at the end